netket/machine/jax.py

```diff
@@ -70,6 +68,31 @@ def _vjp_fun(pars, v, vec, conjugate, forward_fun):
 
             self._vjp_fun = jax.jit(_vjp_fun, static_argnums=(3, 4))
 
+        # R->R
+        elif self._dtype is float and self._outdtype is float:
+
+            grad_fun = jax.jit(jax.grad(forward_scalar))
+            self._perex_grads = jax.jit(jax.vmap(grad_fun, in_axes=(None, 0)))
+
+            def _vjp_fun(pars, v, vec, conjugate, forward_fun):
+                vals, f_jvp = jax.vjp(forward_fun, pars, v.reshape((-1, v.shape[-1])))
+
+                out_r = f_jvp(vec.reshape(vals.shape).real)[0]
+                out_i = f_jvp(-vec.reshape(vals.shape).imag)[0]
+
+                r_flat, tree_fun = tree_flatten(out_r)
+                i_flat, _ = tree_flatten(out_i)
+
+                out_flat = [re + 1j * im for re, im in zip(r_flat, i_flat)]
+
+                if conjugate:
+                    out_flat = [x.conjugate() for x in out_flat]
```

Comment:
I just noticed it Might be a good idea to combine this with the previous Comand, so we sum the real and imaginary part and do the complex conjugation in one operation...

Reply:
Yes good point, I did it also for the other case